### PR TITLE
Tpetra MV: Use specialized KokkosBlas::dot in multiply()

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/tpetra 

## Description
Use a specialized KokkosKernels function in `multiply` when either one of the two factors has only one column.

### CODE REVIEW

**Comment by cgcgcg** [comment]
Hm, does Stokhos not have a `dot` with a matching signature?

**Comment by mhoemmen** [comment]
@cgcgcg wrote:
> Hm, does Stokhos not have a `dot` with a matching signature?

Ah, that's how it works -- Stokhos' PCE specialization of Tpetra has `dot` return `dot_type`, which happens to be `double` and not `scalar_type`.

File: packages/tpetra/core/src/Tpetra_MultiVector_def.hpp [hunk 1]
```diff
@@ -4127,17 +4127,38 @@ namespace Tpetra {
...
-                                    std::make_pair (LO (0), C_numVecs));
       const char ctransA = (transA == Teuchos::NO_TRANS ? 'N' :
                             (transA == Teuchos::TRANS ? 'T' : 'C'));
       const char ctransB = (transB == Teuchos::NO_TRANS ? 'N' :
                             (transB == Teuchos::TRANS ? 'T' : 'C'));
       const impl_scalar_type alpha_IST (alpha);
 
-      ::Tpetra::Details::ProfilingRegion regionGemm ("Tpetra::MV::multiply-call-gemm");
-      KokkosBlas::gemm(&ctransA, &ctransB, alpha_IST, A_sub, B_sub, beta_local, C_sub);
+      // The first two cases are specializations for inner products between a vector and a multivector
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_MultiVector_def.hpp]: `Tpetra::idot` has a specialization for this case.  We should perhaps lift that out into a single function.

**Comment by cgcgcg** [comment]
Replaced by PR #5400

**Pull request closed without merging**